### PR TITLE
Further cleanup of closing sequence

### PR DESCRIPTION
**Pull request by oxtoacart**

### CODE REVIEW

**Comment by myleshorton** [comment]
Looks like I missed this one @oxtoacart. Still relevant?

**Comment by oxtoacart** [comment]
@myleshorton Yup, still relevant, but still not finished from my end :(

**Comment by myleshorton** [comment]
So I should hold off on reviewing right @oxtoacart?

**Comment by oxtoacart** [comment]
@myleshorton Yes, please hold off on reviewing. I'll remove the `[WIP]` from title and ping you when it's ready for a real review.

**Comment by oxtoacart** [comment]
@myleshorton I'm an idiot, I thought this was a different PR. This is okay to review :)

**Comment by myleshorton** [comment]
Cool thanks @oxtoacart!

File: listener.go [hunk 1]
```diff
@@ -79,14 +79,12 @@ func (l *listener) handleConn(conn net.Conn) {
 	l.mx.Unlock()
 
 	defer func() {
+		session.Close()
+		conn.Close()
 		l.mx.Lock()
-		closed := l.closed
+		delete(l.sessions, sessionID)
```
Comment (myleshorton) [inline listener.go:85]: This line certainly seems important @oxtoacart. Is our production code running with this?

Comment (myleshorton) [reply listener.go:85]: Oh I see this was happening just within the `!closed` block -- ignore me.

**Comment by myleshorton** [comment]
OK LGTM @oxtoacart!

**Pull request merged**